src/player.js

```diff
@@ -707,6 +846,48 @@ export default class Player extends FakeEventTarget {
   }
 
   /**
+   * add off text track if there are actual text tracks associated with media
+   * setting this track is the same as calling Player's hideTextTrack
+   * @private
+   * @returns {void}
+   */
+  _addTextTrackOffOption(): void {
+    const textTracks = this.getTracks(TrackTypes.TEXT);
+    if (textTracks && textTracks.length) {
+      this._tracks.push(new TextTrack({
+        active: false,
+        index: textTracks.length,
+        kind: "subtitles",
+        label: "Off",
+        language: "off"
+      }));
+    }
+  }
+
+  _setDefaultTracks(){
+    const activeTracks = this.getActiveTracks();
+    const playbackConfig = this._config.playback;
+
+    if (playbackConfig.textLanguage == "auto"){
+      playbackConfig.textLanguage = Locale.language;
+    }
+
+    this._setDefaultTrack(TrackTypes.TEXT, playbackConfig.textLanguage, activeTracks.text);
+    this._setDefaultTrack(TrackTypes.AUDIO, playbackConfig.audioLanguage, activeTracks.audio);
+  }
+
+  _setDefaultTrack(type: string, language: string, defaultTrack: Track){
+    if (language){
+      const track: ?Track = this._getTracksByType(type).find(track => track.language === language);
+      if (track) {
+        this.selectTrack(track);
+      }
+    } else if (defaultTrack) {
+      this.selectTrack(defaultTrack);
```

Comment:
This is already the selected track - why do we need to select it again?

Reply:
Yes, we don't want to select it necessarily cause we might have config from user that overrides the manifest default.
we only parse it from the adapter and set it as default, but don't actually select it and actual selection is done here where we resolve defaults